src/Fixer/Basic/CurlyBracesPositionFixer.php

```diff
@@ -241,6 +256,27 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
             }
 
             $whitespace = ' ';
+
+            if (self::NEXT_LINE_IF_RETURN_TYPEHINT === $this->configuration[$positionOption]) {
+                $whitespace = $this->whitespacesConfig->getLineEnding().$this->getLineIndentation($tokens, $index);
+
+                $previousTokenIndex = $tokens->getPrevMeaningfulToken($openBraceIndex);
+                $previousToken = $tokens[$previousTokenIndex];
+
+                if ($tokens[$previousTokenIndex]->equals(')')) {
+                    if ($tokens[--$previousTokenIndex]->isComment()) {
+                        --$previousTokenIndex;
+                    }
```

Comment:
Why is this here? Shouldn't we just check if there's `:` + type between `{` (`$openBraceIndex`) and `)`. Any non-meaningful tokens should be skipped. Such decrement looks like optimistic scenario, what if there are 2 comments 😉?

Reply:
/me nods along

I confess, I copied-and-pasted the code for `if (self::NEXT_LINE_UNLESS_NEWLINE_AT_SIGNATURE_END)` (starting on line 280) as the basis for this feature. That original code uses this exact same bit of logic around line 288.

My intuition is that if the logic is good in the original location, it's good probably good here -- unless there is some contextual difference that I'm not seeing. Alternatively, if it's *not* good in the original location, perhaps we have identified a bug.

Reply:
Then we should add test case(s) with comments, like `with comment after opening brace (default)` and `with comment after opening brace (same line)`, to ensure it works correctly.